include/mapbox/variant_visitor.hpp

```diff
@@ -13,7 +15,7 @@ struct visitor<Fn> : Fn
     using type = Fn;
     using Fn::operator();
 
-    visitor(Fn fn) : Fn(fn) {}
+    visitor(Fn&& fn) : Fn(std::forward<Fn>(fn)) {}
```

Comment:
### Improvement
It should be possible to avoid some calls to move ctor using visitor ctor with forwarding references(not rvalue refs.) and decayed types(i.e., `std::decay`) to permit inheritance of Fn and access to Fn::operator(). I believe this is safe here. [This craziness seems to work, BTW, it's only a draft :)](https://gist.github.com/ricardocosme/a84fe41206f5ff849eab91463af75650)